haskell-src/Concordium/Types/ProtocolVersion.hs

```diff
@@ -446,7 +451,7 @@ demoteProtocolVersion = fromSing
 
 -- | Constraint on a type level 'AccountVersion' that can be used to get a corresponding
 --  'SAccountVersion' (see 'accountVersion'). (An alias for 'SingI'.)
-type IsAccountVersion (av :: AccountVersion) = SingI av
+type IsAccountVersion (av :: AccountVersion) = (SingI av, SingI (SupportsValidatorSuspension av))
```

Comment:
For some reason `SingI (SupportsValidatorSuspension av)` isn't inferred. I put it here, because the constraint is used only together with `IsAccountVersion` but there might be a better way to get the instance.

Reply:
So for example, if I drop this, I'll get
 ```Could not deduce ‘SingI (SupportsValidatorSuspension av)’
        arising from a use of ‘put’
      from the context: IsAccountVersion av```
when I'm trying to serialize the field.

Reply:
See e.g. how `putAuthorizations` and `getAuthorizations` handle `asCooldownParameters`: use `mapM_ put _bieAccountIsSuspended` and `_bieAccountIsSuspended <- conditionallyA (sSupportsValidatorSuspension (accountVersion @av)) get`.

Reply:
ok, got it. Removed the constraint.